src/main/java/seedu/allonus/modules/StudyManager.java

```diff
@@ -168,6 +243,30 @@ public Module addModuleParser(String userInput) {
         }
     }
 
+    public void findModule(String userInput) {
+        String moduleKeyword = userInput.replace("find ","");
+        ArrayList<Module> matches = new ArrayList<>();
+        for (Module m: modulesList) {
+            if (m.toString().contains(moduleKeyword)) {
+                matches.add(m);
+            }
+        }
+        if (matches.size() == 0) {
+            printMessage("There are no modules that match \"" + moduleKeyword + "\"");
+        } else {
+            listMatches(matches);
+        }
+    }
+
+    private void listMatches(ArrayList<Module> matches) {
+        System.out.println("    Here are the matching tasks in your list:");
+        int i=1;
```

Comment:
Minor coding standard violation here; consider adding some spaces surrounding the = sign